classes/migration/upgrade/v3_5_0/I9707_WeblateUILocales.php

```diff
@@ -15,18 +15,206 @@
+            ->get()
+            ->pluck('setting_value')
+            ->first();
+        if (in_array($supportedDefaultSubmissionLocale, array_keys($affectedLocales))) {
+            DB::table($this->CONTEXT_SETTINGS_TABLE)
+                ->where('setting_name', '=', 'supportedDefaultSubmissionLocale')
+                ->update(['setting_value' => $affectedLocales[$supportedDefaultSubmissionLocale]]);
+        }
+        $contextLocaleSettingNames = [
+            'supportedFormLocales',
+            'supportedLocales',
+            'supportedSubmissionLocales',
+            'supportedAddedSubmissionLocales',
+            'supportedSubmissionMetadataLocales',
+        ];
+        foreach ($contextLocaleSettingNames as $contextLocaleSettingName) {
+            $contextSettingLocales = DB::table($this->CONTEXT_SETTINGS_TABLE)
+                ->where('setting_name', '=', $contextLocaleSettingName)
+                ->get();
+            foreach ($contextSettingLocales as $contextSettingLocale) {
+                $this->updateArrayLocale($contextSettingLocale->setting_value, $this->CONTEXT_SETTINGS_TABLE, 'setting_value', $contextLocaleSettingName, $this->CONTEXT_COLUMN, $contextSettingLocale->{$this->CONTEXT_COLUMN});
+            }
+        }
+
+
+        // plugin_settings
+        // customBlockManager
+        $blockPluginName = 'customblockmanagerplugin';
+        $blockLocalizedSettingNames = ['blockTitle', 'blockContent'];
+
+        $contextIds = DB::table($this->CONTEXT_TABLE)
+            ->get()
+            ->pluck($this->CONTEXT_COLUMN);
+
+        foreach ($contextIds as $contextId) {
+            $blocks = DB::table('plugin_settings')
+                ->where('plugin_name', '=', $blockPluginName)
+                ->where('setting_name', '=', 'blocks')
+                ->where('context_id', '=', $contextId)
+                ->get()
+                ->pluck('setting_value');
+
+            if (!$blocks->isEmpty()) {
+                $blockNames = $blocks->first();
+
+                $blocksArray = json_decode($blockNames, true);
+                if (is_null($blocksArray)) {
+                    $blocksArray = unserialize($blockNames);
+                }
+
+                foreach ($blocksArray as $block) {
+                    foreach ($blockLocalizedSettingNames as $blockLocalizedSettingName) {
+                        $blockLocalizedContent = DB::table('plugin_settings')
+                            ->where('plugin_name', '=', $block)
+                            ->where('setting_name', '=', $blockLocalizedSettingName)
+                            ->where('context_id', '=', $contextId)
+                            ->first();
+
+                        if (isset($blockLocalizedContent)) {
+                            $this->updateArrayKeysLocaleSetting($blockLocalizedContent->setting_value, 'plugin_settings', 'plugin_setting_id', $blockLocalizedContent->plugin_setting_id);
```

Comment:
I reviewed fast, but it seems to me that this method is doing something similar to the `updateArrayLocale()`.